.github/workflows/docker.yml

```diff
@@ -0,0 +1,89 @@
+name: Docker
+
+on:
+  push:
+    branches: [ master ]
+  
+jobs:
+  build:
+    name: Build and test Waltz - postgresql
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:10.6
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_DB: waltz
+          POSTGRES_PASSWORD: postgres 
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd "pg_isready -U waltz -d waltz"
+          --health-interval 10s 
+          --health-timeout 5s 
+          --health-retries 10
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 8
+      uses: actions/setup-java@v2
+      with:
+        java-version: '8'
+        distribution: 'temurin'
+        cache: 'maven'
+
+    - name: Build with Maven - postgresql
+      run: mvn -B clean package -Pwaltz-postgres,build-postgres,ci -s .build.settings.xml
+
+    - name: Publish build
+      uses: actions/upload-artifact@v2
+      with:
+        name: waltz-web.war
+        path: waltz-web/target/waltz-web.war
+
+    - name: Run containers
+      run: docker-compose up --build -d 
+
+    - name: Health check
+      timeout-minutes: 5
+      run: while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' localhost:8080/waltz-web/)" != "200" ]]; do sleep 5; done
+
+    - name: Verify logs
+      if: always()
+      run: | 
+        docker logs $(docker ps -a | grep '8080->8080' | awk '{print $1}')
+        docker logs $(docker ps -a | grep '5432->5432' | awk '{print $1}')
+
+  publish:
+    name: Push Waltz image - postgresql
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+    - uses: actions/checkout@v2
+    - run: mkdir -p ./waltz-web/target
+    - name: Set up JDK 8
+      uses: actions/setup-java@v2
+      with:
+        java-version: '8'
+        distribution: 'temurin'
+        cache: 'maven'
+
+    - name: Download WAR artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: waltz-web.war
+        path: ./waltz-web/target
+
+    - name: Login to GitHub Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Build and push to GitHub Registry - postgresql
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        push: true
+        tags: ghcr.io/${{ github.repository }}:postgresql 
```

Comment:
1. as discussed above, you probably want to only build here, then run `docker-compose`, then push the image in case of success
2. we should handle some labels in there as well, a bit like we do for Siembol — we should check how waltz does versioning today (tags?) and adapt to their workflow

Reply:
@jgiannuzzi for 2. point, they are using [action softprops]( https://github.com/softprops/action-gh-release) which is basically uploading war file with some script and dependencies
So to follow the same pattern I would prefer using GoRelease for [docker ](https://goreleaser.com/customization/docker/) 
e.g tag would be the actual version and the image name PostgreSQL, as now tag is PostgreSQL and we don't have any versioning

Reply:
indeed they are already using GitHub Actions!

See https://github.com/finos/waltz/blob/master/.github/workflows/maven.yml - you should update your own `master` branch and adapt this PR to integrate with what they already have.

Reply:
yep, I didn't fetch the latest upstream changes for some time, but that's done now, thanks!
With fetching the latest master, I've added that the docker image will be published when `tag` is pushed, that's the same condition they have for their release action step
p.s you'll notice that I changed the image's name so that `tag` could be used as an image tag so it will look like `ghcr.io/ljubon/waltz-postgresql:1.2`
The action was tested on a new branch - https://github.com/ljubon/waltz/actions/runs/1527733596
let me know if you have any feedback about docs [Docker.md](https://github.com/ljubon/waltz/blob/feature/dockerize/docker/DOCKER.md)